File: update_logo.js
const fs = require('fs');
const path = require('path');

const b64Path = path.join(__dirname, 'public', 'logo_b64.txt');
const b64 = fs.readFileSync(b64Path, 'utf8').trim();

// 1. Update src/lib/emailService.ts
const emailServicePath = path.join(__dirname, 'src', 'lib', 'emailService.ts');
let tsContent = fs.readFileSync(emailServicePath, 'utf8');
const tsRegex = /const DEEPWOODS_LOGO_BASE64 = "[^"]+";/;
if (tsRegex.test(tsContent)) {
  tsContent = tsContent.replace(tsRegex, `const DEEPWOODS_LOGO_BASE64 = "${b64}";`);
  fs.writeFileSync(emailServicePath, tsContent, 'utf8');
  console.log('Updated emailService.ts with new logo base64!');
} else {
  console.error('DEEPWOODS_LOGO_BASE64 regex match failed in emailService.ts');
}

// 2. Update google-apps-script/Code.gs
const codeGsPath = path.join(__dirname, 'google-apps-script', 'Code.gs');
let gsContent = fs.readFileSync(codeGsPath, 'utf8');
const gsRegex = /const LOGO_BASE64_DATA = "[^"]+";/;
if (gsRegex.test(gsContent)) {
  gsContent = gsContent.replace(gsRegex, `const LOGO_BASE64_DATA = "${b64}";`);
  fs.writeFileSync(codeGsPath, gsContent, 'utf8');
  console.log('Updated Code.gs with new logo base64!');
} else {
  console.error('LOGO_BASE64_DATA regex match failed in Code.gs');
}

File: update_logo.py
import base64

with open('public/logo.png', 'rb') as f:
    b64 = base64.b64encode(f.read()).decode('utf-8')

# 1. emailService.ts
with open('src/lib/emailService.ts', 'r', encoding='utf-8') as f:
    f1 = f.read()

idx1 = f1.find('const DEEPWOODS_LOGO_BASE64 = "')
if idx1 != -1:
    end1 = f1.find('";', idx1)
    new_f1 = f1[:idx1] + 'const DEEPWOODS_LOGO_BASE64 = "' + b64 + f1[end1:]
    with open('src/lib/emailService.ts', 'w', encoding='utf-8') as f:
        f.write(new_f1)
    print("Updated emailService.ts!")

# 2. Code.gs
with open('google-apps-script/Code.gs', 'r', encoding='utf-8') as f:
    f2 = f.read()

idx2 = f2.find('const LOGO_BASE64_DATA = "')
if idx2 != -1:
    end2 = f2.find('";', idx2)
    new_f2 = f2[:idx2] + 'const LOGO_BASE64_DATA = "' + b64 + f2[end2:]
    with open('google-apps-script/Code.gs', 'w', encoding='utf-8') as f:
        f.write(new_f2)
    print("Updated Code.gs!")
